### DISCUSSION TITLE
How frequently are packages checked on?

### DISCUSSION THREAD

**Discussion by thomasvl** [SwiftPackageIndex/SwiftPackageIndex-Server#1682] [Q&A] [answered]
How frequently are repos checked for updates?

https://swiftpackageindex.com/apple/swift-protobuf says it was updated 20 hours ago.  But clicking through to the build results (https://swiftpackageindex.com/builds/5C7262DB-01F6-4DF7-874D-DD6BECF808F2), it seems to say the build was using aba9915acb9208e8ca5a0e3dcc24690504f86773.  Looking at https://github.com/apple/swift-protobuf/commits/main, there are four newer commits.

So I'm wonder if timing was just off and it hasn't been checked, or if there might be a bug and the build was queued before the commits; but finished after those commits. So some time comparison is being done between the build completion time and thus ignoring the newer commits.

**Comment by daveverwer** [comment, marked as answer]
HI Thomas!

We need to make this more obvious, but the default branch is only built once per 24 hours at maximum. Tagged releases are always built as soon as they are seen, but with people pushing to a default branch we could easily be overwhelmed. There's a small note in the [builds FAQ](https://swiftpackageindex.com/docs/builds#what-revision) but we really need to make it more obvious.

> Note: The build system checks compatibility for every release, no matter how frequent, whereas the system builds the default branch for a package at most every 24-hours. If there are updates to a default branch during the same day, new compatibility checks start once the 24-hour timeout expires.

From looking at the commits on that repo, looks like we happened to scan it for commits just after `aba9915`, around 21 hours ago. So, it should do another `main` build in about 3 hours time, when it will see `9ed8a8e` and build that.

Hope that helps!

**Reply by thomasvl** [reply]
Thanks.  I guess maybe there's a still more to catch up on since SwiftProtobuf doesn't seem to have been updated even after that 24 hour mark.

**Reply by daveverwer** [reply]
I just went to check on them and ironically saw them switch into the queued state. It should be built in the next 5 minutes.

<img width="951" alt="Screenshot 2022-04-08 at 20 55 54@2x" src="https://user-images.githubusercontent.com/5180/162517501-ed34eefc-f13b-477c-8112-fc17edcd8671.png">

**Reply by daveverwer** [reply]
All builds have finished. It built against `9ed8a8e`.

**Comment by finestructure** [comment]
We’re doing a full pass of all packages every couple of hours or so but we coalesce new revisions on default branches over 24h to reduce the number of builds. In practice that means that new releases are turned around shortly after they’re published but builds on default branches will take 24h.

Hope that helps!

**Comment by finestructure** [comment]
Well, what can I say, we just hope it helps 🤣
